sw/device/lib/testing/pwrmgr_testutils.c

```diff
@@ -0,0 +1,29 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/testing/pwrmgr_testutils.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/testing/check.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+void pwrmgr_testutils_enable_low_power(dif_pwrmgr_t *pwrmgr, uint32_t wakeups) {
+  // Enable low power on the next WFI with default settings.
+  // All clocks and power domains are turned off during low power.
+  dif_pwrmgr_domain_config_t config;
+
+  // Issue #6504: USB clock in active power must be left enabled.
+  config = kDifPwrmgrDomainOptionUsbClockInActivePower;
+
+  CHECK(dif_pwrmgr_set_request_sources(pwrmgr, kDifPwrmgrReqTypeWakeup,
+                                       wakeups) == kDifPwrmgrConfigOk);
+  CHECK(dif_pwrmgr_set_domain_config(pwrmgr, config) == kDifPwrmgrConfigOk);
+  CHECK(dif_pwrmgr_low_power_set_enabled(pwrmgr, kDifPwrmgrToggleEnabled) ==
+        kDifPwrmgrConfigOk);
+}
```

Comment:
It will be useful to also add `bool pwrmgr_testutils_is_wakeup_reason(dif_pwrmgr_wakeup_reason_t wakeup_reason)` to abstract lines 46-52 in the test below. It should directly return true if the provided wakeup_reason is the ONLY wakeup reason set.

Reply:
This will also accept multiple reasons "or"ed together. Done.